src/utils/common/common.c

```diff
@@ -160,6 +160,28 @@ char *sstrdup(const char *s) {
   return r;
 } /* char *sstrdup */
 
+char *sstrndup(const char *s, size_t n) {
+  char *r;
+  size_t sz;
+
+  if (s == NULL)
+    return NULL;
+
+  sz = strlen(s);
```

Comment:
Afaik the implementation of strndup uses strnlen and supports also not null-terminated strings. To be in line with that we would need to add sstrnlen probably.

Reply:
Nope, the string must be null terminated as e.g. explained in https://en.cppreference.com/w/c/experimental/dynamic/strndup:

> Returns a pointer to a null-terminated byte string, which contains copies of at most size bytes from the string pointed to by str. If the null terminator is not encountered in the first size bytes, it is added to the duplicated string.

I got the inspiration for the code from gnulib at https://github.com/danielj7/gnulib/blob/master/lib/strndup.c with a replacement of `strnlen()`.

Reply:
Dang, you are right. Thanks for catching this, I'll fix it.